crates/sui-graphql-rpc/src/types/transaction_block.rs

```diff
@@ -297,102 +305,212 @@ impl TransactionBlock {
-                        }
-
-                        if let Some(c) = &filter.after_checkpoint {
-                            query = query.filter(tx::dsl::checkpoint_sequence_number.gt(*c as i64));
-                        }
-
-                        if let Some(c) = &filter.at_checkpoint {
-                            query = query.filter(tx::dsl::checkpoint_sequence_number.eq(*c as i64));
-                        }
-
-                        let before_checkpoint = filter
-                            .before_checkpoint
-                            .map_or(checkpoint_viewed_at + 1, |c| {
-                                c.min(checkpoint_viewed_at + 1)
-                            });
-                        query = query.filter(
-                            tx::dsl::checkpoint_sequence_number.lt(before_checkpoint as i64),
-                        );
-
-                        if let Some(a) = &filter.sign_address {
-                            let sub_query = tx_senders::dsl::tx_senders
-                                .select(tx_senders::dsl::tx_sequence_number)
-                                .filter(tx_senders::dsl::sender.eq(a.into_vec()));
-                            query = query.filter(tx::dsl::tx_sequence_number.eq_any(sub_query));
-                        }
-
-                        if let Some(a) = &filter.recv_address {
-                            let sub_query = tx_recipients::dsl::tx_recipients
-                                .select(tx_recipients::dsl::tx_sequence_number)
-                                .filter(tx_recipients::dsl::recipient.eq(a.into_vec()));
-                            query = query.filter(tx::dsl::tx_sequence_number.eq_any(sub_query));
-                        }
-
-                        if let Some(o) = &filter.input_object {
-                            let sub_query = tx_input_objects::dsl::tx_input_objects
-                                .select(tx_input_objects::dsl::tx_sequence_number)
-                                .filter(tx_input_objects::dsl::object_id.eq(o.into_vec()));
-                            query = query.filter(tx::dsl::tx_sequence_number.eq_any(sub_query));
-                        }
+        let db: &Db = ctx.data_unchecked();
+
+        // Anchor the lower and upper checkpoint bounds if provided from the filter, otherwise
+        // default the lower bound to 0 and the upper bound to `checkpoint_viewed_at`. Increment
+        // `after` by 1 so that we can uniformly select the `min_tx_sequence_number` for the lower
+        // bound. Similarly, decrement `before` by 1 so that we can uniformly select the
+        // `max_tx_sequence_number`. In other words, the range consists of all transactions from the
+        // smallest tx_sequence_number in lo_cp to the max tx_sequence_number in hi_cp.
+        let mut lo_cp = max_option!(
+            filter.after_checkpoint.map(|x| x.saturating_add(1)),
```

Comment:
Making sure I understand: are we doing the add 1 here and sub 1 down in line 343 so that the lo and hi cps are inclusive?

Reply:
yes, from min_tx_sequence_number of lo to max_tx_sequence_number of hi inclusive